news/3931.bugfix.rst

```diff
@@ -0,0 +1 @@
+fix basic auth credentials in --extra-index-urls when using different creds for same domain
```

Comment:
This should describe what use case is fixed exactly instead of what was not working.

Reply:
```suggestion
Prefer credentials from the URL over the previously-obtained credentials from URLs of the same domain, so it is possible to use different credentials on the same index server for different `--extra-index-url` options.
```

This is my understanding to the issue this is fixing. Feel free to rewrite it if you can describe this better; the important thing I want to make sure is to describe the right topic in the changelog.

Reply:
Sure I've updated it.

Reply:
Your suggestion is great. I will update the branch with it.

Reply:
done